robotoff/insights/annotate.py

```diff
@@ -474,6 +475,61 @@ def process_annotation(
         return UPDATED_ANNOTATION_RESULT
 
 
+class UPCImageAnnotator(InsightAnnotator):
+    """Annotation for UPC images
+    Unselect the product images that match a UPC_Image
+    A UPC_Image is defined as an image that has a UPC with a high pct area in the image and thus it is a poor quality main photo
+    """
+
+    @classmethod
+    def process_annotation(
+        cls,
+        insight: ProductInsight,
+        data: Optional[dict] = None,
+        auth: Optional[OFFAuthentication] = None,
+        is_vote: bool = False,
+    ) -> AnnotationResult:
+        product_id = insight.get_product_id()
+        product = get_product(product_id, ["images"])
+
+        image_id = get_image_id(insight.source_image or "")
+
+        if product is None:
+            logger.info(
+                "Product %s not found, cannot unselect image %s", product_id, image_id
+            )
+            return MISSING_PRODUCT_RESULT
+
+        if image_id is None:
+            logger.info("Image %s not found, cannot unselect", image_id)
+            return FAILED_UPDATE_RESULT
+
+        to_unselect = []
+
+        images = product["images"]
+        if image_id not in images:
+            logger.info("Image %s not found in %s", image_id, product_id)
+            return FAILED_UPDATE_RESULT
```

Comment:
We can merge both checks, it's more concise this way.
```suggestion
        images = product["images"]
        if image_id is None or image_id not in images:
            logger.info(
                "Image %s not found for product %s, cannot unselect",
                image_id,
                product_id,
            )
            return FAILED_UPDATE_RESULT

        to_unselect = []
```

Reply:
Done!